drivers/staging/vc04_services/vc-sm-cma/vc_sm.c

```diff
@@ -586,7 +614,7 @@ vc_sm_vpu_event(struct sm_instance *instance, struct vc_sm_result_t *reply,
 	{
 		struct vc_sm_released *release = (struct vc_sm_released *)reply;
 		struct vc_sm_buffer *buffer =
-				(struct vc_sm_buffer *)release->kernel_id;
+					lookup_kernel_id(release->kernel_id);
```

Comment:
I can see this is no less safe than it was before, but shouldn't you acknowledge that the lookup could fail?

Reply:
I've got further patches that add in additional checking (and nearly VPU reloc heap allocations from CMA), and cleans up some of the sequencing on releasing buffers.

There is also the comment immediately below this:
```
/*
		 * FIXME: Need to check buffer is still valid and allocated
		 * before continuing
		 */
```
so I have acknowledged it in comment if not code.